src/realm_dart_sync.cpp

```diff
@@ -138,3 +137,57 @@ RLM_API void realm_dart_sync_on_subscription_state_changed_callback(realm_userda
         (reinterpret_cast<realm_sync_on_subscription_state_changed_t>(ud->dart_callback))(ud->handle, state);
     });
 }
+
+bool wait_dart_before_continue(realm::util::UniqueFunction<void(realm::util::UniqueFunction<void(bool)>*)>* userCallback)
+{
+    std::condition_variable cv;
+    std::mutex m;
```

Comment:
lets use full variable names

Reply:
For such a short function I actually prefer short variable names.

1. The function calls like `notify_one` and `unlock` stand out more.
2. It is easy to see the type anyway, as it is less than 10 lines away.

I also prefer `i` in a short `for` loop.